// @imagina/qintranet/_config/backendPages.js
export default {
  dashboard: {
    permission: null,
    activated: true,
    path: '/intranet/dashboard',
    name: 'qintranet.dashboard',
    page: () => import('@imagina/qintranet/_pages/admin/dashboard'),
    layout: () => import('@imagina/qsite/_layouts/master.vue'),
    title: 'qintranet.sidebar.dashboard',
    icon: 'fas fa-border-all',
    authenticated: true,
    subHeader: {
      refresh: true,
    }
  },
  messenger: {
    permission: null,
    activated: true,
    path: '/intranet/messenger',
    name: 'qintranet.messenger',
    page: () => import('@imagina/qintranet/_pages/admin/messenger'),
    layout: () => import('@imagina/qsite/_layouts/master.vue'),
    title: 'qintranet.sidebar.messenger',
    icon: 'far fa-comments',
    authenticated: true,
    subHeader: {
      refresh: true,
    }
  },
  news: {
    permission: null,
    activated: true,
    path: '/intranet/news',
    name: 'qintranet.news',
    page: () => import('@imagina/qintranet/_pages/admin/news'),
    layout: () => import('@imagina/qsite/_layouts/master.vue'),
    title: 'qintranet.sidebar.news',
    icon: 'far fa-newspaper',
    authenticated: true,
    subHeader: {
      refresh: true,
    }
  },
  newsByCategory: {
    permission: null,
    activated: true,
    path: '/intranet/news/:category',
    name: 'qintranet.news.category',
    page: () => import('@imagina/qintranet/_pages/admin/news'),
    layout: () => import('@imagina/qsite/_layouts/master.vue'),
    title: 'qintranet.sidebar.news',
    icon: 'far fa-newspaper',
    authenticated: true,
    subHeader: {
      refresh: true,
    }
  },
  newsByPost: {
    permission: null,
    activated: true,
    path: '/intranet/news/:category/:slugPost',
    name: 'qintranet.news.post',
    page: () => import('@imagina/qintranet/_pages/admin/post'),
    layout: () => import('@imagina/qsite/_layouts/master.vue'),
    title: 'qintranet.sidebar.news',
    icon: 'far fa-newspaper',
    authenticated: true,
    subHeader: {
      refresh: true,
    }
  },
  directory: {
    permission: null,
    activated: true,
    path: '/intranet/directory',
    name: 'qintranet.directory',
    page: () => import('@imagina/qintranet/_pages/admin/directory'),
    layout: () => import('@imagina/qsite/_layouts/master.vue'),
    title: 'qintranet.sidebar.directory',
    icon: 'fas fa-user-friends',
    authenticated: true,
    subHeader: {
      refresh: true,
    }
  },
  events: {
    permission: null,
    activated: true,
    path: '/intranet/events',
    name: 'qintranet.events',
    page: () => import('@imagina/qintranet/_pages/admin/events'),
    layout: () => import('@imagina/qsite/_layouts/master.vue'),
    title: 'qintranet.sidebar.events',
    icon: 'fas fa-calendar-alt',
    authenticated: true,
    subHeader: {
      refresh: true,
    }
  },

}
